week_4/quiz-trivia-app/src/app/storage.service.ts

```diff
@@ -0,0 +1,28 @@
+export class StorageService {
+  // Save data to localStorage
+  saveData(key: string, data: any): void {
+    if (data === undefined || data === null) {
+      localStorage.removeItem(key);
+    } else {
+      // Save objects as JSON, others (like strings) as simple strings
+      if (typeof data === 'object') {
+        localStorage.setItem(key, JSON.stringify(data)); // Save objects as JSON
+      } else {
+        localStorage.setItem(key, String(data)); // Save strings and primitives as simple strings
+      }
+    }
+  }
+
+  // Get data from localStorage
+  getData<T>(key: string): T | null {
```

Comment:
instead of a generic type, you know the type of data you are saving in storage, create a type for it

Reply:
done